src/gluonts/model/statsforecast.py

```diff
@@ -0,0 +1,232 @@
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import List, Optional, Type
+
+import numpy as np
+
+from statsforecast.models import (
+    ADIDA,
+    AutoARIMA,
+    AutoCES,
+    CrostonClassic,
+    CrostonOptimized,
+    CrostonSBA,
+    ETS,
+    IMAPA,
+    TSB,
+)
+
+from gluonts.core.component import validated
+from gluonts.dataset import DataEntry
+from gluonts.dataset.util import forecast_start
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.model.forecast import QuantileForecast
+
+
+@dataclass
+class ModelConfig:
+    quantile_levels: Optional[List[float]] = None
+    forecast_keys: List[str] = field(init=False)
+    statsforecast_keys: List[str] = field(init=False)
+    intervals: Optional[List[int]] = field(init=False)
+
+    def __post_init__(self):
+        self.forecast_keys = ["mean"]
+        self.statsforecast_keys = ["mean"]
+        if self.quantile_levels is None:
+            self.intervals = None
+            return
+
+        intervals = set()
+
+        for quantile_level in self.quantile_levels:
+            interval = round(
+                200 * (max(quantile_level, 1 - quantile_level) - 0.5)
+            )
+            intervals.add(interval)
+            side = "hi" if quantile_level > 0.5 else "lo"
+            self.forecast_keys.append(str(quantile_level))
+            self.statsforecast_keys.append(f"{side}-{interval}")
+
+        self.intervals = sorted(intervals)
+
+
+class StatsForecastPredictor(RepresentablePredictor):
+    """
+    A predictor type that wraps models from the `statsforecast`_ package.
+
+    This class is used via subclassing and setting the ``ModelType`` class
+    attribute to specify the ``statsforecast`` model type to use.
+
+    .. _statsforecast: https://github.com/Nixtla/statsforecast
+
+    Parameters
+    ----------
+    prediction_length
+        Prediction length for the model to use.
+    quantile_levels
+        Optional list of quantile levels that we want predictions for.
+        By default this is ``None``, giving only the mean predition.
+    **kwargs
+        Keyword arguments to be passed to the model type for construction.
+        The specific arguments accepted or required depend on the
+        ``ModelType``; please refer to the documentation of ``statsforecast``
+        for details.
+    """
+
+    ModelType: Type
+
+    @validated()
+    def __init__(
+        self,
+        prediction_length: int,
+        quantile_levels: Optional[List[float]] = None,
+        **kwargs,
```

Comment:
It might be better to not use `**kwargs` but something like `model_params`. That way the two sets of parameters are separated more clearly.

It's also what we do for prophet, where it's called `prophet_params`.